Danki2/Assets/Scripts/Actor/Mortal/Mortal.cs

```diff
@@ -33,5 +37,30 @@ protected override void Update()
         }
     }
 
+    private void UpdateHealthBar()
+    {
+        if (HealthBar == null || HealthBar.canvas == null)
+        {
+            return;
+        }
+
+        if (Health <= 0)
+        {
+            HealthBar.canvas.enabled = false;
+            return;
+        }
+
+        // Rotate the health bar to face the camera, but lock the y-rotation
+        var healthBarPosition = HealthBar.canvas.transform.position;
+        var cameraPosition = Camera.main.transform.position;
+        var lookAtPosition = 2 * healthBarPosition - cameraPosition;
+        lookAtPosition.x = healthBarPosition.x;
+        HealthBar.canvas.transform.LookAt(lookAtPosition);
+
+        // Update the remaining health display
+        float maxHealth = GetStat(Stat.MaxHealth);
+        HealthBar.transform.localScale = new Vector3(Health / maxHealth, 1f, 1f);
```

Comment:
This logic all looks good to me. I think it would be better though if this didn't live on Mortal, instead you could create a component that lives on the FloatingHealthBar, that requires an `Image` and a `Mortal`. Then this logic can be moved to the Update event on that component and the mortal can be dragged over in the editor (the `Image` from HealthbarFront just needs to be dragged over once in the prefab, you could even make this field private and stop serializing it once it's set for the first time, to stop people using the prefab from setting it themselves?). I think the Mortal class shouldn't really have to worry about the display of the health bar (it's a bit weird that we need the null check at the top of this method and the Player prefab just leaves the Image empty)

Reply:
Yeah I think that would be a good change - since every healthbar has a mortal, but not vice versa

Reply:
You seem have a good instinct for where you should and shouldn't have components